plugin.yml

```diff
@@ -10,6 +10,12 @@ configuration:
   properties:
     server:
       type: string
+    auth-method:
+      type: string
+    role-id:
+      type: string
+    secret-id:
+      type: string
```

Comment:
I believe that we should add dependencies on these two for `auth-method` (I don´t think that they make sense otherwise)... and if they are mandatory when it is it should be codified as well here

Reply:
I've refactored this so that `role-id` and `method` is part of the `auth` object, like so:

```configuration:
  properties:
    server:
      type: string
    auth:
      type: object
      properties:
        method:
          enum:
            - 'approle'
            - ''
        role-id:
          type: string
  required:
    - server
    - auth
  additionalProperties: false
  dependencies:
    auth: [ method ]
```
and updated refs in the auth method to match. I also removed `secret-id` since we shouldn't be passing that as part of the approle auth method, the client should be responsible for setting that before-hand, though I'm not sure the best way to communicate that so any suggestions are appreciated.

Reply:
Makes sense... in that case I would just add a parameter to the `auth` dictionary called something like `secret-variable-name` that you use in the code to get the name of a variable that holds the value to use. I believe I have seen something similar in other plugins that would strike a balance between having a hardcoded variable name and allow customization in the use of the plugin however they decide to provide that sensitive value.

Reply:
@toote good suggestion, I've added it as `secret-env` and it will default to use `VAULT_SECRET_ID` if no value is passed

Reply:
looked at the code and it looks almost ready as you are referencing the value correctly... but not using it anywhere :(

Reply:
fixed!